test/SlowTests/Issues/RavenDB-16614.cs

```diff
@@ -0,0 +1,179 @@
+            {
+                session.Delete("users/arava");
+                await session.SaveChangesAsync();
+            }
+
+            using (var session = store.OpenAsyncSession(new SessionOptions { TransactionMode = TransactionMode.ClusterWide }))
+            {
+                var arava = await session.LoadAsync<User>("users/arava");
+                Assert.Null(arava);
+                var guard = await session.Advanced.ClusterTransaction.GetCompareExchangeValueAsync<object>("rvn-atomic-guard-users/arava");
+                Assert.Null(guard);
+            }
+        }
+
+
+        [Fact]
+        public async Task CanModifyDocumentAfterFirstTime()
+        {
+            var leader = await CreateRaftClusterAndGetLeader(3);
+            using var store = GetDocumentStore(new Options { Server = leader, ReplicationFactor = 3 });
+
+            using (var session = store.OpenAsyncSession(new SessionOptions { TransactionMode = TransactionMode.ClusterWide }))
+            {
+                await session.StoreAsync(new User { Name = "arava" }, "users/arava");
+                await session.StoreAsync(new User { Name = "phoebe" }, "users/phoebe");
+                await session.SaveChangesAsync();
+            }
+
+            using (var session = store.OpenAsyncSession(new SessionOptions { TransactionMode = TransactionMode.ClusterWide }))
+            {
+                var arava = await session.LoadAsync<User>("users/arava");
+                arava.Name = "Arava Eini";
+                var phoebe = await session.LoadAsync<User>("users/phoebe");
+                phoebe.Name = "Phoebe Eini";
+                await session.SaveChangesAsync();
+            }
+        }
+
+        [Fact]
+        public async Task ModificationInAnotherTransactionWillFail()
+        {
+            var leader = await CreateRaftClusterAndGetLeader(3);
+            using var store = GetDocumentStore(new Options { Server = leader, ReplicationFactor = 3 });
+
+            using (var session = store.OpenAsyncSession(new SessionOptions { TransactionMode = TransactionMode.ClusterWide }))
+            {
+                await session.StoreAsync(new User { Name = "arava" }, "users/arava");
+                await session.StoreAsync(new User { Name = "phoebe" }, "users/phoebe");
+                await session.SaveChangesAsync();
+            }
+
+            using (var session = store.OpenAsyncSession(new SessionOptions { TransactionMode = TransactionMode.ClusterWide }))
+            {
+                var user = await session.LoadAsync<User>("users/arava");
+                user.Name = "Arava Eini";
+                var user2 = await session.LoadAsync<User>("users/phoebe");
+                user2.Name = "Phoebe Eini";
+
+                using (var conflictedSession = store.OpenAsyncSession(new SessionOptions {TransactionMode = TransactionMode.ClusterWide}))
```

Comment:
How can we make sure that this session is hitting another node? Or is this done implicit by the test runner?
If implicit, how can we make sure the test also succeeds with hitting one node each time?

Reply:
We aren't trying to do that. Cluster Wide transactions are going to go to the leader anyway, so it doesn't matter from our perspective.